test/integration/consul-container/libs/service/helpers.go

```diff
@@ -121,3 +133,48 @@ func CreateAndRegisterStaticClientSidecar(
 
 	return clientConnectProxy, nil
 }
+
+func GetEnvoyConfigDump(port int, filter string) (string, error) {
+	client := cleanhttp.DefaultClient()
+	url := fmt.Sprintf("http://localhost:%d/config_dump?%s", port, filter)
+
+	res, err := client.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+// GetSidecarCertificate perform a GET to /certs endpoint and returns response body and any error
+func GetSidecarCertificate(port int) (string, error) {
```

Comment:
I think this function can be replaced by calling `GetEnvoyOutput(port, "certs", nil)`